src/hir/mod.rs

```diff
@@ -12,8 +13,22 @@ pub use hir::*;
 /// This function leverages `crate::syntax::parse` and `translator::Translator::translate`
 /// to hide away most of the complexity of `bulloak`'s internal compiler.
 pub fn translate(tree: &str) -> anyhow::Result<Hir> {
-    let ast = crate::syntax::parse(tree)?;
+    let asts = crate::syntax::parse(tree)?;
+    let combiner = combiner::Combiner::new();
+    match combiner.verify(&asts) {
+        Err(_e) => {
+            // @follow-up - propagate this error
+        }
+        _ => {}
+    }
+    let mut hirs = Vec::new();
     let mut discoverer = crate::scaffold::modifiers::ModifierDiscoverer::new();
-    let modifiers = discoverer.discover(&ast);
-    Ok(translator::Translator::new().translate(&ast, modifiers))
+    let translator = translator::Translator::new();
+    for ast in &asts {
+        // @follow-up - use `map` instead of `for _ in`, here and elsewhere?
+        let modifiers = discoverer.discover(ast).unwrap();
+        hirs.push(translator.clone().translate(ast, modifiers));
```

Comment:
```suggestion
        hirs.push(translator.translate(ast, modifiers));
```

Let's try to avoid cloning here by changing https://github.com/alexfertel/bulloak/blob/c38e7795c6a018019ed2e2e0712868b5db0202be/src/hir/translator.rs#L30 to accept `&self` instead of `self`.